connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorPluginsResourceTest.java

```diff
@@ -399,19 +430,59 @@ public void testConnectorPluginsIncludesTypeAndVersionInformation() throws Excep
         );
     }
 
-    protected static ConnectorPluginInfo newInfo(Class<? extends Connector> klass, String version) {
-        return new ConnectorPluginInfo(new MockConnectorPluginDesc(klass, version));
+    @Test
+    public void testListAllPlugins() {
+        Set<ConnectorPluginInfo> connectorPlugins = new HashSet<>(connectorPluginsResource.listConnectorPlugins(false));
+
+        List<Set<MockConnectorPluginDesc<?>>> allPlugins = Arrays.asList(
+                SINK_CONNECTOR_PLUGINS,
+                SOURCE_CONNECTOR_PLUGINS,
+                CONVERTER_PLUGINS,
+                HEADER_CONVERTER_PLUGINS,
+                TRANSFORMATION_PLUGINS,
+                PREDICATE_PLUGINS);
+        for (Set<MockConnectorPluginDesc<?>> plugins : allPlugins) {
+            for (MockConnectorPluginDesc<?> plugin : plugins) {
+                boolean contained = connectorPlugins.contains(newInfo(plugin));
+                if ((ConnectorPluginsResource.CONNECTOR_EXCLUDES.contains(plugin.pluginClass())) ||
+                        (ConnectorPluginsResource.TRANSFORM_EXCLUDES.contains(plugin.pluginClass()))) {
+                    assertFalse(contained);
+                } else {
+                    assertTrue(contained);
+                }
+            }
+            verify(herder, atLeastOnce()).plugins();
+        }
```

Comment:
Same thought w/r/t performing assertions on the complete set of returned plugins:
```suggestion
        Set<Class<?>> excludes = Stream.of(
                    ConnectorPluginsResource.SINK_CONNECTOR_EXCLUDES,
                    ConnectorPluginsResource.SOURCE_CONNECTOR_EXCLUDES,
                    ConnectorPluginsResource.TRANSFORM_EXCLUDES
                ).flatMap(Collection::stream)
                .collect(Collectors.toSet());
        Set<ConnectorPluginInfo> expectedConnectorPlugins = Stream.of(
                    SINK_CONNECTOR_PLUGINS,
                    SOURCE_CONNECTOR_PLUGINS,
                    CONVERTER_PLUGINS,
                    HEADER_CONVERTER_PLUGINS,
                    TRANSFORMATION_PLUGINS,
                    PREDICATE_PLUGINS
                ).flatMap(Collection::stream)
                .filter(p -> !excludes.contains(p.pluginClass()))
                .map(ConnectorPluginsResourceTest::newInfo)
                .collect(Collectors.toSet());

        Set<ConnectorPluginInfo> actualConnectorPlugins = new HashSet<>(connectorPluginsResource.listConnectorPlugins(false));

        assertEquals(expectedConnectorPlugins, actualConnectorPlugins);

        verify(herder, atLeastOnce()).plugins();
```